### DISCUSSION TITLE
Referential actions using SetDefault not working on db push

### DISCUSSION THREAD

**Discussion by gmcamposano** [prisma/prisma#21766] [Q&A]
### Question

Before I had delete on cascade on Clients - so whenever a user was gone, also his clients. Now i want to preserve the clients even though i delete a user.

I modified the schema to look like this and then `prisma db push`, but i still haven´t done any migrations as i am testing it locally / prototyping.  Even though i made the following changes, i still get userId_fkey error.
 
```prisma
model User {
  id               String             @id @default(cuid())
  name             String?
  clients Clients[]
}
model Clients {
  id          String   @id @default(cuid())
  name        String
  userId      String?  @default("deletedUser")
  user        User?    @relation(fields: [userId], references: [id], onDelete: SetDefault)
}
```
```
Foreign key constraint failed on the field: `PosiblesClientes_userId_fkey (index)`
   ...{
  code: 'P2003',
  clientVersion: '5.4.1',
  meta: { field_name: 'PosiblesClientes_userId_fkey (index)' }
}
```
Am i doing something wrong here? There is already data in the DB..

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by gmcamposano** [comment]
any help here?

**Reply by ludralph** [reply]
Hi @gmcamposano 👋 

I have taken a look at your question and can confirm that referential actions using `setDefault` indeed throw the error you reported. I also looked at the [example](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-actions#setdefault) in our documentation, and I got the same error. I'll check in with the engineering team to confirm this behaviour.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
